knimin/lib/data_access.py

```diff
@@ -2399,64 +2400,205 @@ def getProjectNames(self):
+        ----------
+        field : str
+            Field name, i.e., table name under schema "pm"
+        name : str
+            Name of the option
+
+        Returns
+        -------
+        int
+            Id of the option
+
+        Raises
+        ------
+        ValueError
+            If name is not found in the table
+        """
+        sql = "SELECT {} FROM {} WHERE name = %s".format(field + '_id',
+                                                         'pm.' + field)
+        sql_args = [name]
+        id = self._con.execute_fetchone(sql, sql_args)
+        if id is None:
+            raise ValueError('"%s" is not a pre-defined option for %s.'
+                             % (name, field))
+        return id[0]
+
+    def create_sample(self, sample_ids):
+        """Creates samples by id
+
+        Parameters
+        ----------
+        sample_ids : list of str
+            Sample ids to create
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """INSERT INTO pm.sample (sample_id) VALUES (%s)"""
+        self._con.executemany(sql, [[x] for x in sample_ids])
+        return True
+
+    def delete_sample(self, sample_ids):
+        """Delete samples by id
+=======
     def get_plate_info(self, plate_id):
         """Gets attributes of a plate by id
+>>>>>>> 089a92844b668dcbfa9abd240295fc484482dc3d
 
         Parameters
         ----------
-        plate_id : int
-            Plate id
+        sample_ids : list of str
+            Sample ids to delete
 
         Returns
         -------
-        dict
-            Plate attributes
+        bool
+            True if successful
         """
-        sql = """SELECT * FROM pm.plate WHERE plate_id = %s"""
-        return self._con.execute_fetchdict(sql, [plate_id])[0]
+        sql = """DELETE FROM pm.sample WHERE sample_id = %s"""
+        self._con.executemany(sql, [[x] for x in sample_ids])
+        return True
+
+    def create_plate(self, name, email, plate_type):
+        """Creates a new plate and sets values for mandatory fields
+
+        Parameters
+        ----------
+        name : str
+        email : str
+        plate_type : str
+
+        Returns
+        -------
+        int
+            Id of the created plate
+        """
+        sql = """INSERT INTO pm.plate (name, email, plate_type_id)
+                 VALUES (%s, %s, (SELECT plate_type_id FROM pm.plate_type
+                 WHERE name = %s)) RETURNING plate_id"""
+        sql_args = [name, email, plate_type]
+        return self._con.execute_fetchone(sql, sql_args)[0]
+
+    def delete_plate(self, plate_ids):
+        """Deletes plates and corresponding plate-to-sample maps
+
+        Parameters
+        ----------
+        plate_ids : list of int
+            Plate ids to delete
+
+        Returns
+        -------
+        int
+            Number of plates deleted
+        """
+        sql = """DELETE FROM pm.plate_sample WHERE plate_id = %s"""
```

Comment:
Not sure I understand. What about just issuing begin/commit, and just issue the statements direct against the connection object since TRN isn't in this repo? cc @josenavas